### PR TITLE
feat:Added StepInput component

### LINKED ISSUE #768: Need InputGroup Step Up/Down Number Component
### Description
With the refactor of InputGroup, the input group with a number type and step up/down options with callbacks has been omitted directly from InputGroup.  
- This needs to be rebuilt as it's own separate component. 

To consider: the specs have changed for step up/down - see Fiori 3 step Input https://wiki.wdf.sap.corp/wiki/pages/viewpage.action?pageId=2048477546

image
image

### Versions
**fundamental-react:** v0.7.0-rc.75 
**fundamental-styles:** v0.3.0-rc.14

### PR DESCRIPTION
**Pull request by singh-mahendra**
### Description
Added Step input component for numbers.

fixes #768

### CODE REVIEW

File: src/StepInput/StepInput.js [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+    disableStyles,
+    placeholder,
+    readOnly,
+    validationState,
+    value,
+    ...rest
+}, ref) => {
+    useEffect(() => {
+        if (!disableStyles) {
+            require('fundamental-styles/dist/input.css');
```
Comment (jbadan) [inline src/StepInput/StepInput.js]: We can remove this because `FormInput` will bring input.css with it

File: src/StepInput/StepInput.js [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+        </InputGroup.Addon>
+    );
+
+    return (
+        <>
+            <InputGroup className={inputGroupClasses} disabled={disabled}
+                ref={ref} {...rest}
+                onKeyDown={onKeyDownInput}>
+                {readOnly ? null : minusBtn}
+                <FormInput disabled={disabled} onChange={onChangeInputValue}
```
Comment (jbadan) [inline src/StepInput/StepInput.js]: Can you pass `disableStyles={disableStyles}` to InputGroup and FormInput

File: src/StepInput/StepInput.js (hunk 1 above)
Comment (singh-mahendra) [reply src/StepInput/StepInput.js]: Done.Please review.

File: src/StepInput/StepInput.js (hunk 2 above)
Comment (singh-mahendra) [reply src/StepInput/StepInput.js]: Done.Please review.

**Comment by jbadan** [comment]
This looks good to me - do you mind adding some storybook stories following this pattern - https://github.com/SAP/fundamental-react/blob/master/src/InfoLabel/__stories__/InfoLabel.stories.js#L11

File: src/StepInput/StepInput.js [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+                event.preventDefault();
+                event.stopPropagation();
+                break;
+            default:
+        }
+    });
+
+    const minusBtn = (
+        <InputGroup.Addon isButton>
+            <Button disabled={disabled}
```
Comment (jbadan) [inline src/StepInput/StepInput.js]: Can you add an aria-label here with some "Step down" and "Step up" defaults - using the `localizedText` prop like seen here: https://github.com/SAP/fundamental-react/blob/e3e6c9ae7986085a4ce65b39d2da853db258fd06/src/Counter/Counter.js#L36

File: src/StepInput/StepInput.Component.js [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+export const StepInputComponent = () => {
+    return (
+        <ComponentPage
+            description={'The **Step Input** allows numbers to be entered.'}
+            sourceModulePath={path.join(__dirname, './StepInput')}
+            title='Step Input'>
+
+            <Example
+                centered
+                description={'The Input with is typically used to enter numbers.The value can be increased or reduced with the provided controls.'}
```
Comment (jbadan) [inline src/StepInput/StepInput.Component.js]: Small typo - "The StepInput is typically used"

File: src/StepInput/StepInput.js (hunk 3 above)
Comment (singh-mahendra) [reply src/StepInput/StepInput.js]: Added,please review.

File: src/StepInput/StepInput.Component.js (hunk 4 above)
Comment (singh-mahendra) [reply src/StepInput/StepInput.Component.js]: Corrected.Please review.

**Pull request merged**